docs/src/main/asciidoc/_building.adoc

```diff
@@ -1,5 +1,26 @@
 == How to Build Spring Cloud Contract
 
+=== Cloning the repository on Windows
+
+While cloning this project on Windows, some files in the git repository may exceed the Windows maximum file path limit of 255 characters, which may
+result in an incorrectly (probably partially) checked out repository.
+
+To resolve this issue, you can set the `core.longPaths` attribute to `true` or clone the Spring Cloud Contract repository.
+
+
+The following example shows how to set the `core.longPaths` attribute to `true`:
+
+```
+git config --system core.longpaths true
+```
+
+The following example shows how to clone Spring Cloud Contract:
+
+```
+git clone -c core.longPaths=true https://github.com/spring-cloud/spring-cloud-contract.git
+```
+
+
```

Comment:
@Buzzardo @shanman190 Sure! I like the idea of options myself. Will update this shortly.

Reply:
@ankurshukla10, let's wait for Buzzardo's opinion first. I'm kind of curious if we don't just take a simplified approach as what is typically done in git documentation and just use the system example only leaving it to the user to perform an alternative at their own preference.

Reply:
I generally favor more detail rather than less. However, this document isn't about how to configure Git, so we need to not get carried away. That said, I would explain that there are three options and make them into an unnumbered (bulleted) list, thus:

To set the `core.longPaths` attribute to `true`, you have three options:

- Change it for all users of the machine (doing so requires administrator privileges):
+
====
[source,bash]
----
git config --system core.longPaths true
git clone https://github.com/spring-cloud/spring-cloud-contract.git
----
====
- Change it for the current user (no administrative privileges required):
+
====
[source,bash]
----
git config --global core.longPaths true
git clone https://github.com/spring-cloud/spring-cloud-contract.git
----
====

- Change for just this repository (administrative privileges depend on where the repository is being cloned to):
+
====
[source,bash]
----
git clone -c core.longPaths true https://github.com/spring-cloud/spring-cloud-contract.git
----
====

Reply:
It turned the code fences into headings. Sorry about that.

Reply:
@Buzzardo @shanman190  The changes are made, please review and let me know.

Reply:
Please add the code fences. I know it seems like a lot of structure for not much benefit, but it makes all the code listings be consistent. If we use the three backticks approach, these code listings look different than our other code listings. (This all happens because some code listings have callouts, and we have to add this structure to make the code listings with callouts look the same as the listings that don't have callouts. Sorry about the extra work.

Reply:
@Buzzardo 
Just to understand it correctly, am I supposed to remove the back-ticks and use the [source,bash], so now this becomes

[source,bash]
git config --system core.longPaths true
git clone https://github.com/spring-cloud/spring-cloud-contract.git

Reply:
Github formatting letting us down. Let's see if I can make a code block to show the Asciidoc structure:

```
====
[source,bash]
----
git config --system core.longPaths true
git clone https://github.com/spring-cloud/spring-cloud-contract.git
----
====
```

And then do the same for each of the three. Again, I know it's a lot of format for not much code, but it does solve our consistency problem so that all the code listings look the same. Thanks.

Reply:
@Buzzardo  Thanks, made those changes.

Reply:
I have approved the changes I requested. Thanks much.